src/lib/asn1/asn1_obj.h

```diff
@@ -19,17 +19,24 @@ class BER_Decoder;
 class DER_Encoder;
 
 /**
-* ASN.1 Type and Class Tags
+* ASN.1 Class Tags
 */
-enum class ASN1_Tag : uint32_t {
-   UNIVERSAL        = 0x00,
-   APPLICATION      = 0x40,
-   CONTEXT_SPECIFIC = 0x80,
+enum class ASN1_Class : uint32_t {
+    UNIVERSAL        = 0b0000'0000,
+    APPLICATION      = 0b0100'0000,
+    CONTEXT_SPECIFIC = 0b1000'0000,
+    PRIVATE          = 0b1100'0000,
 
-   CONSTRUCTED      = 0x20,
+    CONSTRUCTED      = 0b0010'0000,
```

Comment:
I wrote the class values as binary literals because it was easier for me to understand what was going on. If you want, I can convert them to hex literals.

Reply:
Yes for this binary is better (it wasn't binary before since until this past November we were still C++11-only)